### PR TITLE
Prevent sending more ERC20 tokens than you have

### LINKED ISSUE #1829: ERC20 does not guard against sending more than your balance

### PR DESCRIPTION
**Pull request by James-Sangalli**
Fixes #1829

### CODE REVIEW

File: AlphaWallet/Transfer/ViewControllers/SendViewController.swift [hunk 1]
```diff
@@ -213,7 +213,7 @@ class SendViewController: UIViewController, CanScanQRCode {
         let input = targetAddressTextField.value.trimmed
         guard let address = AlphaWallet.Address(string: input) else { return displayError(error: Errors.invalidAddress) }
         let amountString = amountTextField.ethCost
-        let parsedValue: BigInt? = {1
```
Comment (James-Sangalli) [inline AlphaWallet/Transfer/ViewControllers/SendViewController.swift:216]: @hboon no idea why this was here

Comment (hboon) [reply AlphaWallet/Transfer/ViewControllers/SendViewController.swift:216]:
```
git log -S "let parsedValue: BigInt? = {1"

commit 9f44b5612040d18760b51e55db56e90956a87acb
Author: Hwee-Boon Yar <hboon@motionobj.com>
Date:   2020-02-26 12:52:56 +0800

    Add event support for TokenScript schema 2020/03
```

:p

Funny thing is it still works.

**Comment by hboon** [comment]
@James-Sangalli Tests broken elsewhere. Will fix it.

File: AlphaWallet/Transfer/ViewControllers/SendViewController.swift [hunk 2]
```diff
@@ -229,12 +229,17 @@ class SendViewController: UIViewController, CanScanQRCode {
                 return EtherNumberFormatter.full.number(from: amountString, decimals: token.decimals)
             }
         }()
-        guard let value = parsedValue else {
+
+        guard let value = parsedValue, value > 0 else {
             return displayError(error: SendInputErrors.wrongInput)
         }
 
         if case .nativeCryptocurrency = transferType, let balance = session.balance, balance.value < value {
```
Comment (hboon) [inline AlphaWallet/Transfer/ViewControllers/SendViewController.swift:237]: Can you make it into a switch instead?

**Comment by hboon** [comment]
Test suite fixed in #1842

File: AlphaWallet/Transfer/ViewControllers/SendViewController.swift (hunk 2 above)
Comment (James-Sangalli) [reply AlphaWallet/Transfer/ViewControllers/SendViewController.swift]: @hboon sussed

**Pull request merged**